tests/Doctrine/Tests/ORM/Query/SelectSqlGenerationTest.php

```diff
@@ -1929,7 +1929,7 @@ public function testQuotedWalkJoinVariableDeclaration(): void
     {
         $this->assertSqlGeneration(
             'SELECT u, a FROM Doctrine\Tests\Models\Quote\User u JOIN u.address a',
-            'SELECT q0_."user-id" AS userid_0, q0_."user-name" AS username_1, q1_."address-id" AS addressid_2, q1_."address-zip" AS addresszip_3, q1_.type AS type_4 FROM "quote-user" q0_ INNER JOIN "quote-address" q1_ ON q0_."address-id" = q1_."address-id" AND q1_.type IN (\'simple\', \'full\')'
+            'SELECT q0_."user-id" AS userid_0, q0_."user-name" AS username_1, q1_."address-id" AS addressid_2, q1_."address-zip" AS addresszip_3, q1_.type AS type_4 FROM "quote-user" q0_ INNER JOIN "quote-address" q1_ ON q0_."user-id" = q1_."user-id" AND q1_.type IN (\'simple\', \'full\')'
```

Comment:
Why does that change? Are we using an association from the non-owning side in this test?

Can we change this test so that it still covers configuration read from a `@JoinColumn` annotation, maybe using the other side of the association?

Reply:
> Are we using an association from the non-owning side in this test?

Yes I think we are. Here, we are establishing a join with `address-id`, which should not even be allowed to exist in the first place IMO.

> Can we change this test so that it still covers configuration read from a @JoinColumn annotation, maybe using the other side of the association?

I'm not sure what exactly you suggest, or if it should even be allowed. Is it valid to have such an annotation when not on the owning side?

Reply:
BTW, if the goal of this test is to ensure the columns used to perform the join are quoted, that's the case, so…

Reply:
When you remote the `@JoinColumn` from `User::$address`, which is the inverse side of the association (since it uses `mappedBy`), and we all think that the inverse side does not matter at all... _why_ does the SQL change then?